app/scripts/controllers/token-rates.js

```diff
@@ -26,32 +26,22 @@ class TokenRatesController {
   async updateExchangeRates () {
     if (!this.isActive) { return }
     const contractExchangeRates = {}
-    // copy array to ensure its not modified during iteration
-    const tokens = this._tokens.slice()
-    for (const token of tokens) {
-      if (!token) return log.error(`TokenRatesController - invalid tokens state:\n${JSON.stringify(tokens, null, 2)}`)
-      const address = token.address
-      contractExchangeRates[address] = await this.fetchExchangeRate(address)
+    const pairs = this._tokens.map(token => `pairs[]=${token.address}/ETH`)
+    const query = pairs.join('&')
+    if (this._tokens.length > 0) {
+      try {
+        const response = await fetch(`https://exchanges.demo.balanc3.net/pie?${query}&autoConversion=true`)
```

Comment:
What's the demo part of this URL? Is this a temporary API?

Reply:
Looking at https://github.com/MetaMask/metamask-extension/issues/5741#issuecomment-438067768 we should be able to drop it?